frontend/middleware_validate.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+const (
+	wantedResourceProviderNamespace = "Microsoft.RedHatOpenShift"
+	resourceTypeHCPOpenshiftCluster = "HcpOpenShiftClusters"
+)
+
+var rxResourceGroupName = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+
+func MiddlewareValidate() MiddlewareFunc {
```

Comment:
Since we have multiple middleware performing validation, can we rename this one to be `MiddlewareValidateStatic`?